common_states/states/common_states/locate_object/locate_object_ESCGUK/script.py

```diff
@@ -0,0 +1,23 @@
+
+def execute(self, inputs, outputs, gvm):
+    fetch = gvm.get_variable("robot")
+    rospy = gvm.get_variable("rospy")
+    head = fetch[4]
+    grasp_cli = fetch[3]
+    torso = fetch[5]
+    look = inputs["look"]
+    head.look_at("map", look["x"], look["y"], look["w"])
+    self.logger.info("looked at object")
+    self.logger.info("raising torso")
+    torso.set_height(.4)
+    while not rospy.is_shutdown():
```

Comment:
Some use cases probably require control over how many times reperception happens, possibly with other actions happening before retrying (e.g. moving the head), so it doesn't make sense to retry forever. I would make `retries` an int parameter of the state and have it default to 3, that way it can be used with no retries in other situations.